application/analyst/services/chapter_indexing_service.py

```
@@ -62,6 +66,13 @@ async def ensure_collection(self, novel_id: str) -> None:
         """
         collection_name = self._get_collection_name(novel_id)
 
+        existing = await self._vector_store.list_collections()
+        # 如果 legacy collection 存在，保留使用旧名，避免读写分歧
+        legacy_name = f"novel_{novel_id}_chunks"
+        if legacy_name in existing:
+            logger.debug(f"Using legacy collection: {legacy_name}")
```

Ruff (0.15.10)

[error] Undefined name `logger`

(F821)